packages/transport/src/transports/abstractApi.ts

```diff
@@ -187,51 +193,40 @@ export abstract class AbstractApiTransport extends AbstractTransport {
                 }
                 const { path } = getPathBySessionResponse.payload;
 
-                try {
-                    const protocol = customProtocol || v1Protocol;
-                    const bytes = buildMessage({
-                        messages: this.messages,
-                        name,
-                        data,
-                        encode: protocol.encode,
-                    });
-                    const chunks = createChunks(
-                        bytes,
-                        protocol.getChunkHeader(bytes),
-                        this.api.chunkSize,
-                    );
-                    const apiWrite = (chunk: Buffer) => this.api.write(path, chunk, signal);
-                    const sendResult = await sendChunks(chunks, apiWrite);
-                    if (!sendResult.success) {
-                        throw new Error(sendResult.error);
```

Comment:
this is the main difference, I already had typed error, but used throw statement here, which leads to loss of type information which is again band-aided on line 228.